main.rb

```diff
@@ -1,102 +1,139 @@
+    my_each { |i| selected_items.push(i) if yield(i) }
+    selected_items
+  end
+
+  # rubocop:disable  Metrics/CyclomaticComplexity, Metrics/PerceivedComplexity
+  def my_all?(pattern = nil)
+    result = true
+    if block_given?
+      my_each { |i| result &= (yield i) }
+    elsif block_given? && !pattern.nil?
+      my_each { |i| result &= pattern == i }
+    else
+      my_each { |i| result &= i }
     end
+    result
+  end
 
-    def my_any?
-      raise LocalJumpError, 'You have not given a block to the method' unless block_given?
-      result = false
-      self.my_each do |e|
-        unless yield(e)
-          result = true
-          break
-        end
-      end
-      result
+  def my_any?(pattern = nil)
+    if block_given?
+      my_each { |i| return true if yield(i) }
+    elsif pattern.class == Class
+      my_each { |i| return true if i.class == pattern }
+    elsif pattern.class == Regexp
+      my_each { |i| return true if i =~ pattern }
+    elsif pattern.nil?
+      my_each { |i| return true if i }
+    else
+      my_each { |i| return true if i == pattern }
     end
+    false
+  end
 
-    def my_none?
-      raise LocalJumpError, 'You have not given a block to the method' unless block_given?
-      return true if !self.my_any?{|x| yield x}
+  def my_none?(pattern = nil)
+    if block_given?
+      my_each { |i| return false if yield(i) }
+    elsif pattern.class == Class
+      my_each { |i| return false if i.class == pattern }
+    elsif pattern.class == Regexp
+      my_each { |i| return false if i =~ pattern }
+    elsif pattern.nil?
+      my_each { |i| return false if i }
+    else
+      my_each { |i| return false if i == pattern }
     end
-    
-    def my_count
-      self.size
+    true
+  end
+
+  def my_inject(*args)
+    arr = to_a.dup
+    if args[0].nil?
+      operand = arr.shift
+    elsif args[1].nil? && !block_given?
+      symbol = args[0]
+      operand = arr.shift
+    elsif args[1].nil? && block_given?
+      operand = args[0]
+    else
+      operand = args[0]
+      symbol = args[1]
     end
 
-    # NB: Commented because it was amended in Step 8 of the project
-
-    # def my_map
-    #   raise LocalJumpError, 'You have not given a block to the method' unless block_given?
-    #   mapped = []
-    #   self.my_each {|x|mapped.push(yield(x))}
-    #   mapped
-    # end
-
-    def my_map(arg = nil)
-      raise LocalJumpError, 'You have not given a block to the method' unless block_given?
-      arr = []
-      my_each do |i|
-        if !arg.nil?
-          arr.push(arg.call(i))
-        else
-          arr.push(yield(i))
-        end
-      end
-      arr
+    arr[0..-1].my_each do |i|
+      operand = if symbol
+                  operand.send(symbol, i)
+                else
+                  yield(operand, i)
+                end
     end
+    operand
+  end
+  # rubocop:enable  Metrics/CyclomaticComplexity, Metrics/PerceivedComplexity
+
+  def my_count
+    size
+  end
```

Comment:
**[REQUIRED]**
* Counts the number of items in enum that are equal to item if an argument is given.
```
Failure/Error: expect(array.my_count(LOWEST_VALUE)).to eq array.count(LOWEST_VALUE)
     
     ArgumentError:
       wrong number of arguments (given 1, expected 0)
```
* Counts the number of elements yielding a true value if a block is given.
```
Failure/Error: expect(array.my_count(&block)).to eq array.count(&block)
     
       expected: 46
            got: 100
```